students/rick_halsell/lesson02/exercisetwo.py

```diff
@@ -0,0 +1,133 @@
+    full_line_func()
+    for i in range(4):
+        pipe_space_func()
+    full_line_func()
+    return
+# calling the function
+box_func()
+print()
+
+# part 2
+# Printing a Grid with size options
+# function to print the top line of the box
+print('Part 2')
+def plusfunc():
+    plussymbol = [print('+', end=' ') for plussymbol in range(1)]
+    return
+# function to print the second line of the box
+def tabfunc():
+    tabsymbol = [print('-', end=' ') for tabsymbol in range(1)]
+    return
+def pipefunc():
+    pipesymbol = [print('|', end=' ') for pipesymbol in range(1)]
+    return
+def spacefunc():
+    spacesymbol = [print(' ' * ((size * 2)-1) , end=' ') for spacesymbol in range(1)]
+    return
+# function to call the functions to create the box
+def box_size_func(size):
+    # function to print the second line of the box
+    def pipe_space_func():
+        num_pipe = [ print('|', end=' ') for num_pipe in range(1)]
+        num_space = [print(' ' * ((size * 2)-1) , end=' ') for num_space in range(1)]
+        num_pipe = [ print('|', end=' ') for num_pipe in range(1)]
+        num_space = [print(' ' * ((size * 2)-1), end=' ') for num_space in range(1)]
+        num_pipe = [ print('|', end=' ') for num_pipe in range(1)]
+        print()
+        return
+    plusfunc() # insert plus symbol
+    for i in range(size): # insert dash symbol
+        tabfunc()
+    plusfunc() # insert plus symbol
+    for i in range(size): # insert dash symbol
+        tabfunc()
+    plusfunc() # insert plus symbol
+    print()
+    for i in range(size): # instert pipe symbol
+        pipe_space_func()
+    plusfunc() # insert plus symbol
+    for i in range(size): # insert dash symbol
+        tabfunc()
+    plusfunc() # insert plus symbol
+    for i in range(size): # insert dash symbol
+        tabfunc()
+    plusfunc() # insert plus symbol
+    print()
+    for i in range(size): # instert pipe symbol
+        pipe_space_func()
+    plusfunc() # insert plus symbol
+    for i in range(size): # insert dash symbol
+        tabfunc()
+    plusfunc() # insert plus symbol
+    for i in range(size): # insert dash symbol
+        tabfunc()
+    plusfunc() # insert plus symbol
+    print()
+    return
+# calling the function
+box_size_func(7)
+
+# part 3
+# Printing a Grid with units options
+# function to print the top line of the box
+print('Part 3')
+# function to call the functions to create the box
+# I need some help with trying to use to following function in the while loop to produce the
+# grid.
+# I'm receiving this error: TypeError: can't multiply sequence by non-int of type 'str'
+# I understand what the function is saying, I need to know how to work around this error.
+
+def plusfunc(units,size):
+    #plussymbol = [print('+', ((* units * 2)), end=' ') for plussymbol in range(1)]
+    #plussymbol = '+ ' + ('- ' * units)
+    #print(plussymbol)
+    plus = '+ ' + ('- ' * units)
+    pipe = '| ' + (' '* (2 * size))
+    print(plus * pipe + '+')
```

Comment:
Getting an error here: TypeError: can't multiply sequence by non-int of type 'str'